File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/core/key/ttl/SwlTtlKeyManager.js
/**
 * @return {SwlTtlKeyManager}
 * @interface {SwlTtlKeyManager}
 * @extends {SwlKeyManager}
 */
function SwlTtlKeyManager() {

}


/**
 * @return {boolean}
 */
SwlTtlKeyManager.prototype.preferSetAndTtl = function () {

}

/**
 *
 * @param selfAsymSign {String}
 * @param keyPair {AsymKeyPair}
 * @param ttlSeconds {long}
 * @return {void}
 */
SwlTtlKeyManager.prototype.setSelfKeyPairWithTtl = function (selfAsymSign,keyPair,ttlSeconds) {

}

/**
 *
 * @param otherAsymSign {String}
 * @param publicKey {String}
 * @param ttlSeconds {long}
 * @return {void}
 */
SwlTtlKeyManager.prototype.setOtherPublicKeyWithTtl = function (otherAsymSign,publicKey,ttlSeconds) {

}

/**
 *
 * @param selfAsymSign {String}
 * @param ttlSeconds {long}
 */
SwlTtlKeyManager.prototype.setSelfTtl = function (selfAsymSign,ttlSeconds) {

}

/**
 *
 * @param otherAsymSign {String}
 * @param ttlSeconds {long}
 */
SwlTtlKeyManager.prototype.setOtherTtl = function (otherAsymSign,ttlSeconds) {

}

export default SwlTtlKeyManager

File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/core/key/ttl/SwlTtlKeyManager.java
package i2f.swl.core.key.ttl;

import i2f.jce.std.encrypt.asymmetric.key.AsymKeyPair;
import i2f.swl.core.key.SwlKeyManager;

/**
 * @author Ice2Faith
 * @date 2024/8/8 22:14
 * @desc
 */
public interface SwlTtlKeyManager extends SwlKeyManager {
    boolean preferSetAndTtl();

    void setSelfKeyPairWithTtl(String selfAsymSign, AsymKeyPair keyPair, long ttlSeconds);

    void setOtherPublicKeyWithTtl(String otherAsymSign, String publicKey, long ttlSeconds);

    void setSelfTtl(String selfAsymSign, long ttlSeconds);

    void setOtherTtl(String otherAsymSign, long ttlSeconds);
}
